corehq/apps/couch_sql_migration/tests/test_migration.py

```diff
@@ -566,6 +565,23 @@ def test_timings(self):
         for t_stat in tracked_stats:
             self.assertTrue(any(r_stat.startswith(t_stat) for r_stat in received_stats))
 
+    def test_dry_run(self):
+        self.assertFalse(should_use_sql_backend(self.domain_name))
+        call_command(
+            'migrate_domain_from_couch_to_sql',
+            self.domain_name,
+            MIGRATE=True,
+            no_input=True,
+            dry_run=True
+        )
+        clear_local_domain_sql_backend_override(self.domain_name)
+        with self.assertRaises(AssertionError):
```

Comment:
This seems overly broad since `AssertionError` could come from an unexpected place (the same applies to `SystemExit` raised by `ArgumentParser.error`).

Suggestion: use [`assertRaisesRegexp`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp) to validate the expected error message (and provide a meaningful error message when the error occurs).

Reply:
If it's coming from an unexpected place, I'd generally assume another test would catch that, but I guess that's a bad assumption